lte/gateway/c/session_manager/LocalEnforcer.cpp

```diff
@@ -1748,72 +1748,30 @@ void LocalEnforcer::process_rules_to_install(
     std::vector<DynamicRuleInstall> dynamic_rule_installs,
     RulesToProcess& rules_to_activate, RulesToProcess& rules_to_deactivate,
     SessionStateUpdateCriteria& uc) {
-  std::time_t current_time     = std::time(nullptr);
-  std::string ip_addr          = session.get_config().common_context.ue_ipv4();
-  std::string ipv6_addr        = session.get_config().common_context.ue_ipv6();
-  const std::string session_id = session.get_session_id();
-  for (const auto& rule_install : static_rule_installs) {
-    const auto& id = rule_install.rule_id();
-    if (session.is_static_rule_installed(id)) {
-      // Session proxy may ask for duplicate rule installs.
-      // Ignore them here.
-      continue;
-    }
-    PolicyRule static_rule;
-    if (!rule_store_->get_rule(id, &static_rule)) {
-      MLOG(MERROR) << "static rule " << id
-                   << " is not found, skipping install...";
-      continue;
-    }
+  RulesToSchedule to_schedules;
+  session.process_rule_installs(
+      static_rule_installs, dynamic_rule_installs, rules_to_activate,
+      rules_to_deactivate, to_schedules, uc);
 
-    RuleLifetime lifetime(rule_install);
-    if (lifetime.activation_time > current_time) {
-      session.schedule_static_rule(id, lifetime, uc);
+  const std::string session_id = session.get_session_id();
```

Comment:
stylistic: I think you can catch this in const std::string &? the tmp return value will be bound to the lifetime of session_id. Though I believe modern compilers do this for you automatically, not sure.

Reply:
Yeah I think better practice to use const ref